hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/AbstractScheduleHandler.java

```diff
@@ -16,75 +16,361 @@
+import com.hedera.hapi.node.base.ResponseCodeEnum;
+import com.hedera.hapi.node.base.ScheduleID;
+import com.hedera.hapi.node.base.Timestamp;
+import com.hedera.hapi.node.base.TransactionID;
+import com.hedera.hapi.node.state.schedule.Schedule;
 import com.hedera.hapi.node.transaction.TransactionBody;
-import com.hedera.node.app.spi.UnknownHederaFunctionality;
+import com.hedera.node.app.service.schedule.ReadableScheduleStore;
+import com.hedera.node.app.service.schedule.ScheduleRecordBuilder;
+import com.hedera.node.app.spi.signatures.SignatureVerification;
+import com.hedera.node.app.spi.workflows.HandleContext;
+import com.hedera.node.app.spi.workflows.HandleException;
 import com.hedera.node.app.spi.workflows.PreCheckException;
 import com.hedera.node.app.spi.workflows.PreHandleContext;
-import com.hedera.node.app.spi.workflows.PreHandleDispatcher;
+import com.hedera.node.app.spi.workflows.TransactionKeys;
+import com.hedera.node.app.spi.workflows.record.SingleTransactionRecordBuilder;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
 import java.util.Objects;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
+import java.util.Set;
+import java.util.concurrent.ConcurrentSkipListSet;
+import java.util.function.Function;
+import java.util.stream.Collectors;
 
 /**
  * Provides some implementation support needed for both the {@link ScheduleCreateHandler} and {@link
  * ScheduleSignHandler}.
  */
 abstract class AbstractScheduleHandler {
-    private static final Logger logger = LogManager.getLogger(AbstractScheduleHandler.class);
-    public static final String IGNORED_EXCEPTION_MESSAGE =
-            "Ignored inner context pre check exception and replaced with UNRESOLVABLE_REQUIRED_SIGNERS.";
+    protected static final String NULL_CONTEXT_MESSAGE =
+            "Dispatcher called the schedule delete handler with a null context; probable internal data corruption.";
+
+    protected static void markRecordSuccess(
+            @NonNull final ScheduleID currentScheduleId,
+            @NonNull final Schedule currentSchedule,
+            @NonNull final HandleContext context) {
+        final ScheduleRecordBuilder recordBuilder = context.recordBuilder(ScheduleRecordBuilder.class);
+        // set the schedule ID for THIS transaction
+        recordBuilder.scheduleID(currentScheduleId);
+        // set the child transaction ID for THIS transaction
+        recordBuilder.scheduledTransactionID(ScheduleUtility.transactionIdForScheduled(currentSchedule));
+        // If all went well, we return SUCCESS.
+        // @note the interface below should always be implemented by all record builders
+        if (recordBuilder instanceof SingleTransactionRecordBuilder base) {
+            base.status(ResponseCodeEnum.SUCCESS);
+        }
```

Comment:
If `recordBuilder` really should always implement `SingleTransactionRecordBuilder`, would it be better to throw an error when we encounter a record builder that doesn't? I.e. instead of an if statement:
```java
// throws ClassCastException for an invalid record builder
((SingleTransactionRecordBuilder) recordBuilder).status(ResponseCodeEnum.SUCCESS);  
```

Reply:
ClassCastException is an ugly way to fail.  If we want things to fail (and it's not clear we do), a HandleException would be better (and could be in the else).  Something to consider for the next PR that adds more testing and cleans up anything found in testing.

Reply:
Agreed, `ClassCastException` is an ugly way to fail. It sounded like a better idea yesterday 🥴 Though I do think we should indicate there is an error somehow, whether that's throwing an exception (as you added) or logging an error. Maybe the appropriate error code is `FAIL_INVALID`? Not sure. Anyway, as long as we indicate something went wrong so we can fix it, that's fine with me